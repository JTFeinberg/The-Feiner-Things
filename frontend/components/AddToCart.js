import React, { Component } from 'react'
import { Mutation } from 'react-apollo'
import gql from 'graphql-tag'

export default class AddToCart extends Component {
    render() {
        const { id } = this.props
        return (
            <button type="button">
                Add To Cart ðŸ›’
            </button>
        )
    }
}
