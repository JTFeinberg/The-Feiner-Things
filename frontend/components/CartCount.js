import React from 'react'
import PropTypes from 'prop-types'
import { TransitionGroup, CSSTransition } from 'react-transition-group'
import styled from 'styled-components'

const Dot = styled.div`
  background: ${props => props.theme.red};
  color: white;
  border-radius: 50%;
  padding: 0.5rem;
  line-height: 2rem;
  min-width: 3rem;
  margin-left: 1rem;
  font-weight: 100;
  /* These two lines standardize the changing widths of different numbers */
  font-feature-settings: 'tnum';
  font-variant-numeric: tabular-nums;
`

const CartCount = ({ count }) => (
  <TransitionGroup>
    <CSSTransition
      unmountOnExit
      className="count"
      classNames="count"
      key={count}
      timeout={{ enter: 4000, exit: 4000 }}>
      <Dot>{count}</Dot>
    </CSSTransition>
  </TransitionGroup>
)

export default CartCount
