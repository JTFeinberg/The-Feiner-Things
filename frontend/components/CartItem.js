import React, { Component } from 'react'
import formatMoney from '../lib/formatMoney'
import styled from 'styled-components'
import PropTypes from 'prop-types'

const CartItemStyles = styled.li``

const CartItem = ({ cartItem }) => <CartItemStyles>{cartItem.id}</CartItemStyles>

CartItem.propTypes = {
  cartItem: PropTypes.object.isRequired
}

export default CartItem
