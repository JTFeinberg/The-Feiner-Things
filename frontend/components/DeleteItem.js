import React, { Component } from 'react'

export default class DeleteItem extends Component {
  render() {
    return <button>Delete Item</button>
  }
}
