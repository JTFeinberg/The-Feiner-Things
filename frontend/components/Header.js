const Header = () => (
  <div>
    <div className="bar">
      <a href="">The Feiner Things</a>
    </div>
    <div className="sub-bar">
      <p>Search</p>
    </div>
    <div>Cart</div>
  </div>
);

export default Header;
