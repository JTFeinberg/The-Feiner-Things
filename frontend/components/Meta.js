const Meta = () => (
<div>META</div>
)

export default Meta