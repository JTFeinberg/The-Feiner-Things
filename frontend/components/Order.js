import React, { Component } from 'react'
import PropTypes from 'prop-types'

export default class Order extends Component {
  static propTypes = {
    id: PropTypes.string.isRequired
  }
  render() {
    return <p>Order ID: {this.props.id}</p>
  }
}
