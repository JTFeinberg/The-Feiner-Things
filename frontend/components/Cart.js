import React from 'react'
import { Query, Mutation } from 'react-apollo'
import gql from 'graphql-tag'
import { adopt } from 'react-adopt'
import User from './User'
import CartItem from './CartItem'
import CartStyles from './styles/CartStyles'
import Supreme from './styles/Supreme'
import CloseButton from './styles/CloseButton'
import SickButton from './styles/SickButton'
import calcTotalPrice from '../lib/calcTotalPrice'
import formatMoney from '../lib/formatMoney'

const LOCAL_STATE_QUERY = gql`
  query {
    # @client directive tells apollo to avoid going to db
    # and instead go to client store for data
    cartOpen @client
  }
`

const TOGGLE_CART_MUTATION = gql`
  mutation {
    toggleCart @client
  }
`

const Composed = adopt({
  user: <User />,
  toggleCart: <Mutation mutation={TOGGLE_CART_MUTATION} />,
  localState: <Query query={LOCAL_STATE_QUERY} />
})

const Cart = () => {
  return (
    <Composed>
      {({ user, toggleCart, localState }) => {
        const { me } = user.data
        if (!me) return null
        const { cart, name } = me
        return (
          <Query query={LOCAL_STATE_QUERY}>
            {({ data }) => (
              <CartStyles open={data.cartOpen}>
                <header>
                  <CloseButton title="close" onClick={toggleCart}>
                    &times;
                  </CloseButton>
                  <Supreme>{name}'s Cart</Supreme>
                  <p>
                    You have {cart.length} item{cart.length > 1 ? 's' : ''} in your cart
                  </p>
                </header>
                <ul>
                  {cart.map(cartItem => (
                    <CartItem key={cartItem.id} cartItem={cartItem} />
                  ))}
                </ul>
                <footer>
                  <p>{formatMoney(calcTotalPrice(cart))}</p>
                  <SickButton>Checkout</SickButton>
                </footer>
              </CartStyles>
            )}
          </Query>
        )
      }}
    </Composed>
  )
}

export default Cart
export { LOCAL_STATE_QUERY, TOGGLE_CART_MUTATION }
