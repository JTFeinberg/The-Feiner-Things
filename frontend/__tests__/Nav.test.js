import React from 'react'
import { mount } from 'enzyme'
import toJSON from 'enzyme-to-json'
import wait from 'waait'
import { MockedProvider } from 'react-apollo/test-utils'
import Nav from '../components/Nav'
import { CURRENT_USER_QUERY } from '../components/User'
import { fakeUser } from '../lib/testUtils'
