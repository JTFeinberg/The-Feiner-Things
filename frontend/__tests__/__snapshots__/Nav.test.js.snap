// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Nav /> redners a minimal nav when signed out 1`] = `
<ul
  className="NavStyles-sc-773yok-0 gJvNUs"
  data-test="nav"
>
  <Link
    href="/items"
  >
    <a
      href="/items"
      onClick={[Function]}
    >
      Shop
    </a>
  </Link>
  <Link
    href="/signup"
  >
    <a
      href="/signup"
      onClick={[Function]}
    >
      Sign In
    </a>
  </Link>
</ul>
`;

exports[`<Nav /> renders the amount of items in the cart 1`] = `
<div
  className="count CartCount__Dot-egg65q-1 jXUeCZ"
>
  9
</div>
`;
